From 314a0ed691ca194e6a35a8a84629e0542de630da Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 13 Oct 2020 20:22:10 +0300
Subject: [PATCH] [SW-12754] gaudi: security: add pcie hbw rrs

To complete the HBW protection add initialization of the PCIe IF RRs on
the areas not protected by the IF RTRs RRs (accessed directly from PCIe
- not through IF).
The PCIe IF RRs are implemented using "min and max" concept, so we can
use a single register to protect all the directly accessible HBW areas.

Change-Id: Iec3b8cf3736e0a15b38cad5f75811615605c6a6a
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 47d9bb7..ccf2149 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1137,6 +1137,9 @@
 	}
 }
 
+#define mmSTM_BASE	0x7FF4000000ULL
+#define mmRAZWI_BASE	0x7FFBFFF000ULL
+
 static void preboot_secure_hbw(void)
 {
 	/* HBM RR_0 0 - 32GB */
@@ -1176,6 +1179,14 @@
 
 	/* PCIe FW SRAM RR_7 0x7F_FBFF_0000 - 0x7F_FBFF_8000 (32KB) */
 	preboot_secure_hbw_rr(7, 0x7FFBFF0000, 0x3FFFFFFF8000);
+
+	/*
+	 * PCIe IF HBW RRs are for direct access from PCIe IF to PSOC domain:
+	 * STM, SPI Flash, Scratchpad SRAM, PCIe FW SRAM.
+	 * Assign the PCIe IF RR_16 (0-15 are allocated for LBW) for this job.
+	 * NOTE: The PCIe RRs use min/max concept and HIT is: min < hit < max
+	 */
+	preboot_secure_pcie_rr(16, mmSTM_BASE - 1, mmRAZWI_BASE);
 }
 
 #define PCIE_WRAP_RRS_NUM	32
